Assets/MixedRealityToolkit-SDK/Features/UX/Scripts/Pointers/BaseControllerPointer.cs

```diff
@@ -190,11 +190,17 @@ public uint PointerId
             }
         }
 
+        private string pointerName = string.Empty;
+
         /// <inheritdoc />
         public string PointerName
         {
-            get { return gameObject.name; }
-            set { gameObject.name = value; }
+            get { return pointerName; }
+            set
+            {
+                pointerName = value;
+                gameObject.name = value;
```

Comment:
This will still null ref if the gameObject is destroyed.

Reply:
Yes, but we only set the name once we create the pointer, so I doubt that can ever really happen.

Reply:
```suggestion
                if(gameObject == null)
                {
                    Debug.LogError($"{pointerName}'s GameObject reference is missing!");
                    return;
                }
                
                gameObject.name = value;
```